org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/EclipseSbtBuildManager.scala

```diff
@@ -122,7 +133,7 @@ class EclipseSbtBuildManager(val project: IScalaProject, settings0: Settings) ex
     clearTasks(added)
     removeFiles(removed)
     sources ++= added
-    runCompiler(sources.asJFiles)
+    runCompiler(bulkFilter(sources.toSet).asJFiles)
```

Comment:
So, now the build manager builds only a subset of all files (corresponding to the build scope). However, there's only one `analysisMap` per builder, is this correctly and completely transitioned to the Sbt incremental compiler? This is pretty tricky to test.. things might be working fine, but I don't think so.

In my tests, I see a lot of full builds. In fact, I'm not sure I have a single incremental build anymore. Judging by log messages, just adding a method in one source file, I get this in the log:

```
Initial source changes: 
	removed:Set(/Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/Base.scala, /Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/JavaBase.java, /Users/dragos/Documents/runtime-New_configuration/base/src/test/scala/base/TestFoo.java, /Users/dragos/Documents/runtime-New_configuration/base/src/test/scala/base/TestScala.scala)
	added: Set()
	modified: Set()
Removed products: Set()
External API changes: API Changes: Set()
Modified binary dependencies: Set()
Initial directly invalidated sources: Set(/Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/Base.scala, /Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/JavaBase.java, /Users/dragos/Documents/runtime-New_configuration/base/src/test/scala/base/TestFoo.java, /Users/dragos/Documents/runtime-New_configuration/base/src/test/scala/base/TestScala.scala)

Sources indirectly invalidated by:
	product: Set()
	binary dep: Set()
	external source: Set()
2015-04-18 15:50:04,593 DEBUG [Worker-34] - EclipseSbtBuildManager - All initially invalidated sources: Set(/Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/Base.scala, /Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/JavaBase.java, /Users/dragos/Documents/runtime-New_configuration/base/src/test/scala/base/TestFoo.java, /Users/dragos/Documents/runtime-New_configuration/base/src/test/scala/base/TestScala.scala)

2015-04-18 15:50:04,594 DEBUG [Worker-34] - EclipseSbtBuildManager - Created transactional ClassfileManager with tempDir = /Users/dragos/Documents/runtime-New_configuration/base/.tmpBin
2015-04-18 15:50:04,594 DEBUG [Worker-34] - EclipseSbtBuildManager - Recompiling all 0 sources: invalidated sources (4) exceeded 50.0% of all sources
2015-04-18 15:50:04,594 DEBUG [Worker-34] - EclipseSbtBuildManager - About to delete class files:
	/Users/dragos/Documents/runtime-New_configuration/base/bin/base/Base.class
	/Users/dragos/Documents/runtime-New_configuration/base/bin/base/JavaBase.class
	/Users/dragos/Documents/runtime-New_configuration/base/bin/base/TestScala.class
```

Then I see another round, where it thinks that sources were added

```
Initial source changes: 
	removed:Set()
	added: Set(/Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/Base.scala, /Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/JavaBase.java)
	modified: Set()
Removed products: Set()
External API changes: API Changes: Set()
Modified binary dependencies: Set()
Initial directly invalidated sources: Set(/Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/Base.scala, /Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/JavaBase.java)

Sources indirectly invalidated by:
	product: Set()
	binary dep: Set()
	external source: Set()
2015-04-18 15:50:04,614 DEBUG [Worker-34] - EclipseSbtBuildManager - All initially invalidated sources: Set(/Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/Base.scala, /Users/dragos/Documents/runtime-New_configuration/base/src/main/scala/base/JavaBase.java)

2015-04-18 15:50:04,614 DEBUG [Worker-34] - EclipseSbtBuildManager - Created transactional ClassfileManager with tempDir = /Users/dragos/Documents/runtime-New_configuration/base/.tmpBin
2015-04-18 15:50:04,614 DEBUG [Worker-34] - EclipseSbtBuildManager - Recompiling all 2 sources: invalidated sources (2) exceeded 50.0% of all sources
```

This can be easily tested with even one project, running in the debugger.